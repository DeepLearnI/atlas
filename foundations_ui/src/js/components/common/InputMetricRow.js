import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CommonActions from '../../actions/CommonActions';

class InputMetricRow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isError: this.props.isError,
      job: this.props.job,
      isMetric: this.props.isMetric,
      allInputMetricColumn: this.props.allInputMetricColumn,
      hiddenInputParams: this.props.hiddenInputParams,
      rowNumber: this.props.rowNumber,
      onMetricRowClick: this.props.onMetricRowClick,
      key: this.props.key,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      isError: nextProps.isError,
      job: nextProps.job,
      isMetric: nextProps.isMetric,
      allInputMetricColumn: nextProps.allInputMetricColumn,
      hiddenInputParams: nextProps.hiddenInputParams,
      rowNumber: nextProps.rowNumber,
      onMetricRowClick: nextProps.onMetricRowClick,
      key: nextProps.key,
    });
  }

  render() {
    const {
      job, isError, isMetric, allInputMetricColumn, hiddenInputParams, rowNumber, key, onMetricRowClick,
    } = this.state;
    const { onClickOpenModalJobDetails } = this.props;
    const cells = CommonActions.getInputMetricCells(job,
      isError,
      isMetric,
      allInputMetricColumn,
      hiddenInputParams,
      rowNumber,
      onClickOpenModalJobDetails);
    return (
      <div
        role="presentation"
        className="job-table-row"
        data-class="job-table-row"
        onClick={() => onMetricRowClick(job, key)}
        onKeyDown={() => onMetricRowClick(job, key)}
      >
        {cells}
      </div>
    );
  }
}

InputMetricRow.propTypes = {
  onMetricRowClick: PropTypes.func,
  key: PropTypes.string,
  job: PropTypes.object,
  isError: PropTypes.bool,
  isMetric: PropTypes.bool,
  allInputMetricColumn: PropTypes.array,
  hiddenInputParams: PropTypes.array,
  rowNumber: PropTypes.number,
  onClickOpenModalJobDetails: PropTypes.func,
};
const defaultFunc = () => console.warn('InputMetricRow: onClick func missing.');
InputMetricRow.defaultProps = {
  onMetricRowClick: defaultFunc,
  key: '',
  job: {},
  isError: false,
  isMetric: false,
  allInputMetricColumn: [],
  hiddenInputParams: [],
  rowNumber: 0,
  onClickOpenModalJobDetails: () => null,
};

export default InputMetricRow;
