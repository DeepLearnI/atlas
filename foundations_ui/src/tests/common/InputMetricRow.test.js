import React from 'react';
import ReactDOM from 'react-dom';
import InputMetricRow from '../../js/components/common/InputMetricRow';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders InputMetricRow', () => {
  const wrapper = shallow(<InputMetricRow/>);
  expect(wrapper).toMatchSnapshot();
});