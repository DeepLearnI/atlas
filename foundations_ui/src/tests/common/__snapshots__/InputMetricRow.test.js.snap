// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders InputMetricRow 1`] = `
<div
  className="job-table-row"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="presentation"
/>
`;
